### ISSUE TITLE
EL7 postfix chroot

### ISSUE THREAD

**Issue by cdenneen** [voxpupuli/puppet-postfix#214]
Currently the following change applies:

```
Info: Applying configuration version '003e0658d144095db0dc546ed0e8c3a4aedb328c'
Notice: /Stage[main]/Postfix::Files/File[/etc/postfix/master.cf]/content:
--- /etc/postfix/master.cf	2018-05-22 14:48:32.461110827 -0400
+++ /tmp/puppet-file20180522-3533-1p8t2rt	2018-05-22 14:49:58.283204671 -0400
@@ -30,7 +30,7 @@
 verify    unix  -       -       -       -       1       verify
 flush     unix  n       -       -       1000?   0       flush
 proxymap  unix  -       -       n       -       -       proxymap
-smtp      unix  -       -       n       -       -       smtp
+smtp      unix  -       -       -       -       -       smtp
 # When relaying mail as backup MX, disable fallback_relay to avoid MX loops
 relay     unix  -       -       -       -       -       smtp
 	-o fallback_relay=

Info: Computing checksum on file /etc/postfix/master.cf
```

but causes issues:
```
May 22 14:50:05 ctcgit01 postfix/postfix-script[10757]: starting the Postfix mail system
May 22 14:50:05 ctcgit01 postfix/master[10759]: daemon started -- version 2.10.1, configuration /etc/postfix
May 22 14:50:05 ctcgit01 postfix/qmgr[10761]: AEE03C01E475: from=<root@ctcgit01.domain.com>, size=480, nrcpt=1 (queue active)
May 22 14:50:05 ctcgit01 postfix/smtp[10768]: fatal: unknown service: smtp/tcp
May 22 14:50:05 ctcgit01 postfix/postfix-script[10785]: refreshing the Postfix mail system
May 22 14:50:05 ctcgit01 postfix/master[10759]: reload -- version 2.10.1, configuration /etc/postfix
May 22 14:50:05 ctcgit01 postfix/qmgr[10797]: AEE03C01E475: skipped, still being delivered
May 22 14:50:06 ctcgit01 postfix/master[10759]: warning: process /usr/libexec/postfix/smtp pid 10768 exit status 1
May 22 14:50:06 ctcgit01 postfix/master[10759]: warning: /usr/libexec/postfix/smtp: bad command startup -- throttling
```

Guessing undef needs to be `n` rather than `-`?

**Comment by cdenneen** [comment]
I'm not sure having to set `postfix::chroot: false` should be required here rather than fixing why it's breaking.
Should there be `/var/spool/postfix/etc` structure created with necessary files?
Is there anyone running this on RedHat/CentOS?

**Comment by cdenneen** [comment]
Without this it’s broken. I’m not near computer to write tests for this. But I consider this kind of a critical bug fix for EL7

**Comment by cdenneen** [comment]
If we can at least get a release for 1.6.1 that has the `chroot => false` parameter then this issue can be fixed later but for now in order to disable chroot we need a new release since 1.6.0

**Comment by raphink** [comment]
Ah right, the chroot feature is not released yet so we wouldn't be breaking any releases if we changed the default behavior. I'm guessing defaulting to `n` is fine, if that works with all OSes.

**Comment by raphink** [comment]
https://github.com/camptocamp/puppet-postfix/pull/224

**Comment by olifre** [comment]
I think this has been fixed by merging #224, no?

**Comment by ekohl** [comment]
I'm going to assume the issue was fixed by #224.
